src/ngraph_builder.cc

```diff
@@ -1969,45 +1979,30 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
             "Number of inputs is not 3 for Slice");
       }
 
-      tf::Node* tf_input;
-      tf::Node* tf_begin;
-      tf::Node* tf_size;
-      TF_RETURN_IF_ERROR(op->input_node(0, &tf_input));
-      TF_RETURN_IF_ERROR(op->input_node(1, &tf_begin));
-      TF_RETURN_IF_ERROR(op->input_node(2, &tf_size));
-
-      auto ng_input = ng_op_map.find(tf_input->name());
-      if (ng_input == ng_op_map.end()) {
-        return tf::errors::InvalidArgument("Missing input: " +
-                                           tf_input->name());
-      }
-      auto ng_begin = ng_op_map.find(tf_begin->name());
-      if (ng_begin == ng_op_map.end()) {
-        return tf::errors::InvalidArgument("Missing input: " +
-                                           tf_begin->name());
-      }
-      auto ng_size = ng_op_map.find(tf_size->name());
-      if (ng_size == ng_op_map.end()) {
-        return tf::errors::InvalidArgument("Missing input: " + tf_size->name());
-      }
+      shared_ptr<ng::Node> ng_input;
+      TRY_FETCH_NG_OP(ng_input, 0);
```

Comment:
Again - once the GetNgOp(...) is properly done, these macros will be replaced by a TF_RETURN_IF_ERROR style.